packages/ember-metal/lib/computed.js

```diff
@@ -257,6 +257,12 @@ ComputedPropertyPrototype.property = function() {
         `Please refactor from \`Ember.computed('${property}', function() {});\` to \`Ember.computed('${property.slice(0, -6)}.[]', function() {})\`.`,
       property.slice(-5) !== '@each'
     );
+    warn(
```

Comment:
If its an assertion it may break people's apps who are using it today without knowing that it doesn't work. :+1: for suggesting an intermediate CP in the meantime.